src/bp/ui-shared/src/FormFields/TagInputList/index.tsx

```diff
@@ -0,0 +1,83 @@
+import { Position, Tooltip } from '@blueprintjs/core'
+import React, { useEffect, useRef, useState } from 'react'
+import { lang } from '~/translations'
+import { controlKey } from '~/utils/keyboardShortcuts'
+import ShortcutLabel from '~/ShortcutLabel'
+
+import AddButton from '../../Contents/Components/Fields/AddButton'
+
+import { SingleTag } from './SingleTag'
+
+export interface Item {
+  name: string
+  synonyms: string[]
+}
+
+const TagInputList = ({ onChange, placeholder, items, addBtnLabel }) => {
+  const [localItems, setLocalItems] = useState(items || [])
+  const focusedElement = useRef(items.length)
+
+  useEffect(() => {
+    setLocalItems(items ?? [])
+  }, [items])
+
+  const updateLocalItem = (index: number, item): void => {
+    if (item.name === '' && item.synonyms.length === 1) {
```

Comment:
very picky comment but normally you should mutate a copy instead (and you copy it anyways line 31)